src/main/java/seedu/placebook/commons/util/StringUtil.java

```diff
@@ -65,4 +67,21 @@ public static boolean isNonZeroUnsignedInteger(String s) {
             return false;
         }
     }
+
+    /**
+     * Returns the String representation of fields of the list, joined by delimiter.
+     *
+     * @param list the list to be converted to String
+     * @param function function to get the field of each item
+     * @param delimiter delimiter to join the String
+     * @param <T> the type of the item in the List
+     * @return the result String
+     */
+    public static <T> String listToString(List<T> list, Function<T, String> function, String delimiter) {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (T t : list) {
+            stringBuilder.append(function.apply(t)).append(delimiter);
+        }
+        return stringBuilder.substring(0, stringBuilder.length() - 2);
```

Comment:
Just noticed this maybe should be -1, because java doesnt take the index at stringBuilder.length() - 2, if im not wrong it might cut out the last letter.

Reply:
I just realised this needs to be `- delimiter.length()`